backend/app/routes.py

```diff
@@ -101,9 +103,30 @@ def get_links_between_two_nodes(from_node_id, to_node_id):
         abort(400, description="Source node can not be the same as target node.")
         return
 
-    connection = getConnection()
+    
+    links = get_link_geom(from_node_id, to_node_id)
 
-    with connection:
+
+    shortest_link_data = {
+        "source": from_node_id, 
+        "target": to_node_id,
+        "links": links,
+        # the following three fields are for compatibility and should eventually be removed
+        "path_name": "",
+        "link_dirs": [ link['link_dir'] for link in links ],
+        "geometry": {
+            "type": "MultiLineString",
+            "coordinates": [ link['geometry']['coordinates'] for link in links ]
+        }
+    }

```

Comment:
Move this to within the `get_links` function

Reply:
WAIT

Reply:
No, actually, I'm reading my comment on line 114 and realizing we should just leave this as is. This is parsing the link lines into a `MultiLineString` (similar to the `GeometryCollection` I mentioned) which will make it harder to associate individual links to their respective line geometries. I was only doing that here for compatibility with the frontend which will change eventually. Sorry for the confusion.